crates/rnote-engine/src/pens/eraser.rs

```diff
@@ -20,15 +20,73 @@ pub enum EraserState {
     Down(Element),
 }
 
+#[derive(Debug, Default, Clone, Copy)]
+pub struct EraserMotion {
+    last_element: Option<(Element, Instant)>,
+    speed: f64,
+    pub added_width: f64,
+}
+
+impl EraserMotion {
+    pub const SMOOTHING_FACTOR_ACCEL: f64 = 3.0;
+    pub const SMOOTHING_FACTOR_DECEL: f64 = 10.0;
+    pub const SPEED_LIMIT: f64 = 10000.0;
+    pub const MIN_SPEED: f64 = 100.0;
+    pub const SPEED_SCALING: f64 = 0.01;
+    pub const SPEED_SCALING_STEP_SIZE: f64 = 25.0;
+
+    fn update(&mut self, element: Element, time: Instant) {
+        if let Some((last_element, last_element_time)) = self.last_element {
+            let delta = element.pos - last_element.pos;
+            let delta_time = time - last_element_time;
```

Comment:
Actually, after looking at my issue (with large speed jumps), it's not impossible to get a delta time that's either too small or negative (and in that case the - clamps the value to zero).

We should only update the speed if the delta_time > 0.0 or superior to an expected minimum time delta.
From https://github.com/flxzt/rnote/blob/7f3ff8aafcaf3137bec9993329667e5e9fbf4c4e/crates/rnote-engine/src/pens/penholder.rs#L497 (for each screen frame, gtk send all the events in one go and we only take events so that they are spread apart by at least the value here), this allows only one update per screen frame, so something like 8ms at 120 Hz.

Maybe a threshold at 4 ms would be good.

This will fix this issue for good